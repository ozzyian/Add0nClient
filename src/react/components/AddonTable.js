import React from 'react';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Table from 'react-bootstrap/Table';
import Button from 'react-bootstrap/Button';
import AddonListItem from './AddonListItem';
import Spinner from 'react-bootstrap/Spinner';

/**
 *
 */
class AddonTable extends React.Component {
  /**
   *
   * @param {*} props
   */
  constructor(props) {
    super(props);
    this.state = {loading: !this.props.initiated, addons: this.props.addons};
    this.showAddons = this.showAddons.bind(this);
  }

  /**
   *
   */
  componentDidMount() {}
  /**
   * @return {*}
   */
  showAddons() {
    return this.state.addons.map((addon) => {
      return <AddonListItem key={addon.id} addon={addon} />;
    });
  }

  /**
   * @return {*}
   */
  render() {
    if (this.state.loading) {
      return (
        <Spinner
          className="center-spinner"
          animation="border"
          role="status"
        ></Spinner>
      );
    } else {
      return (
        <Container id="container" fluid>
          <Row>
            <Button>Update all</Button>
            <Button>Sync with remote</Button>
          </Row>
          <Row>
            <Table responsive striped bordered hover>
              <thead>
                <tr>
                  <th>ID</th>
                  <th>Addon</th>
                  <th>Status</th>
                  <th>Version</th>
                  <th>Author</th>
                </tr>
              </thead>
              <tbody>{this.showAddons()}</tbody>
            </Table>
          </Row>
        </Container>
      );
    }
  }
}

export default AddonTable;
