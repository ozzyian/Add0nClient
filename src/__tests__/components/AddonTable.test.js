import React from 'react';
import {shallow} from 'enzyme';
import AddonTable from '../../react/components/AddonTable';
import fs from 'fs';

describe('Snapshot tests', () => {
  it('renders correctly', () => {
    const component = shallow(<AddonTable />);
    expect(component).toMatchSnapshot();
  });

  it('renders correctly when loading is done', () => {
    const testData = fs.readFileSync(
      './src/__tests__/resources/addon_info.json',
    );
    const testAddon = JSON.parse(testData);
    const component = shallow(<AddonTable />);
    component.setState({loading: false, addons: [testAddon]});
  });
});
