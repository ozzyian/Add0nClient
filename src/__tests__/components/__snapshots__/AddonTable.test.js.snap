// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests renders correctly 1`] = `
<Spinner
  animation="border"
  className="center-spinner"
  role="status"
/>
`;
